.github/workflows/ci-ubuntu.yml

```diff
@@ -32,82 +32,58 @@ on:
   test-cubical:
     runs-on: ubuntu-latest
     steps:
-
-      ########################################################################
-      ## SETTINGS
-      ##
-      ## AGDA_BRANCH picks the branch of Agda to use to build the library.
-      ########################################################################
-
-      - name: Initialise variables
-        run: |
-          # Pick Agda version
-          echo "AGDA_BRANCH=v2.6.3" >> $GITHUB_ENV
-
-
-      ########################################################################
-      ## CHECKOUT
-      ########################################################################
-
-      # By default github actions do not pull the repo
-      - name: Checkout cubical
-        uses: actions/checkout@v3
-
-      ########################################################################
-      ## CACHING
-      ########################################################################
-
       # This caching step allows us to save a lot of building time by only
       # downloading ghc and cabal, rebuilding Agda, and re-checking unchanged
       # library files if absolutely necessary i.e. if we change either the
       # version of Agda, ghc, or cabal that we want to use for the build.
-      - name: Cache cabal packages
-        uses: actions/cache@v3
-        id: cache-cabal
+      - name: Restore external dependencies cache
+        uses: actions/cache/restore@v3
+        id: cache-external-restore
         with:
-          path: |
-            ~/.cabal/packages
-            ~/.cabal/store
-            ~/.cabal/bin
+          path: ${{ env.CACHE_PATHS }}
           key: ${{ runner.os }}-${{ env.GHC_VERSION }}-${{ env.CABAL_VERSION }}-${{ env.AGDA_BRANCH }}
 
-      - name: Cache library build
-        uses: actions/cache@v3
-        id: cache-library
-        with:
-          path: ./_build
-          key: library-${{ runner.os }}-${{ env.GHC_VERSION }}-${{ env.CABAL_VERSION }}-${{ env.AGDA_BRANCH }}-${{ hashFiles('Cubical/**') }}
-          restore-keys: |
-            library-${{ runner.os }}-${{ env.GHC_VERSION }}-${{ env.CABAL_VERSION }}-${{ env.AGDA_BRANCH }}-
-
       ########################################################################
       ## INSTALLATION STEPS
       ########################################################################
 
       - name: Install cabal
-        if: steps.cache-cabal.outputs.cache-hit != 'true'
+        if: steps.cache-external-restore.outputs.cache-hit != 'true'
         uses: haskell/actions/setup@v2
         with:
           ghc-version: ${{ env.GHC_VERSION }}
           cabal-version: ${{ env.CABAL_VERSION }}
 
-      - name: Put cabal programs in PATH
-        run: echo "~/.cabal/bin" >> $GITHUB_PATH
-
       - name: Cabal update
         run: cabal update
 
       - name: Download and install Agda from github
-        if: steps.cache-cabal.outputs.cache-hit != 'true'
+        if: steps.cache-external-restore.outputs.cache-hit != 'true'
```

Comment:
Just a comment (not on your change, but on this CI design):
Since Agda is only rebuilt if the cache key changes, you could not have a movable `AGDA_BRANCH` identifier like `master`.
To facilitate the latter, one would first have to resolve `ADGA_BRANCH` to a SHA and then use this as cache key (component).
```sh
SHA=$(git rev-parse ${{ env.AGDA_BRANCH }})
```
Of course, this has the drawback that one has to checkout the Agda repo _always_ (and before cache restoration) just to do this resolution.

Reply:
Maybe we should use GraphQL action to query the latest hash on `AGDA_BRANCH`?

Reply:
If you want released versions of Agda, there is now https://github.com/wenkokke/setup-agda as the official Agda installer (should be moved to the `agda` organization at some point).

Re GraphQL: You mean https://github.com/octokit/graphql-action ?

Reply:
Yes I meant this action, but I've missed the elephant: GitHub also has a classic REST API interface.